docs/safe-typeorm.md

```diff
@@ -0,0 +1,345 @@
+# Safe-TypeORM
+
+- [Outline](#outline)
+- [Setup](#setup)
+  - [NPM Package](#npm-package)
+  - [Prerequisites](#prerequisites)
+- [Features](#features)
+  - [JoinQueryBuilder](#joinquerybuilder)
+  - [JsonSelectBuilder](#jsonselectbuilder)
+  - [InsertCollection](#insertcollection)
+- [Guide Documents](#guide-documents)
+
+## Outline
+
+![JoinQueryBuilder](https://raw.githubusercontent.com/samchon/safe-typeorm/master/assets/demonstrations/join-query-builder.gif)
+
+Enhance type safety, therefore make below things possible:
+
+  - When writing **SQL query**,
+    - Errors would be detected in the **compilation level**
+    - **Auto Completion** would be provided
+    - **Type Hint** would be supported
+  - When **SELECT**ing for **JSON** transformation
+    - **App-Join** with the related entities would be automatically done
+    - Exact JSON **type** would be **automatically deduced**
+    - The **performance** would be **automatically tuned**
+
+I've published [safe-typeorm](https://github.com/samchon/safe-typeorm) to suggest TypeORM to adapt TMP concept, and demonstrate how it works (related issue: [typeorm#9868](https://github.com/typeorm/typeorm/issues/9868)). I hope [safe-typeorm](https://github.com/samchon/safe-typeorm) to be adpated by `TypeORM`, therefore become key features of next `v0.4` update.
+
+However, this is a third party package and is not managed by the entirety of the `TypeORM` core team yet. Thus, until this [safe-typeorm](https://github.com/samchon/safe-typeorm) be accepted by `TypeORM` core team, please report any issues found with the library in the [appropriate repository](https://github.com/samchon/safe-typeorm).
+
+
+
+
+
+## Setup
+
+### NPM Package
+
+```bash
+npm install --save typeorm@0.2
```

Comment:
Does Safe-TypeORM work with the latest 0.3 version?

Reply:
Do not support `typeorm@0.3` yet.

It is very simple thing for me to support the `v0.3` (only one hour required).

However. as `safe-typeorm` use custom decorators for relationship, most new features of v0.3 would be disabled.

Reply:
it doesn't make sense to support 0.2 anymore. Officially it's not maintained by us anymore, and `v0.3` is one year old already!

Reply:
Okay, I'll make it support on Sunday. 

And may I send a PR to `typeorm` about some convenient features of `safe-typeorm`? 

I think that `JsonSelectBuilder` can be easily implemented.

Reply:
no, we need to make TypeORM lightweight instead of adding new features.